huaweicloud/services/dds/resource_huaweicloud_dds_lts_log.go

```diff
@@ -0,0 +1,338 @@
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	createDdsLtsLogOpt.JSONBody = utils.RemoveNil(buildCreateDdsLtsLogBodyParams(d))
+
+	instanceID := d.Get("instance_id").(string)
+
+	retryFunc := func() (interface{}, bool, error) {
+		createDdsLtsLogResp, err := createDdsLtsLogClient.Request("POST", createDdsLtsLogPath, &createDdsLtsLogOpt)
+		retry, err := handleMultiOperationsError(err)
+		return createDdsLtsLogResp, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     ddsInstanceStateRefreshFunc(createDdsLtsLogClient, instanceID),
+		WaitTarget:   []string{"normal"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error associating DDS with LTS log: %s", err)
+	}
+
+	d.SetId(instanceID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"false"},
+		Target:     []string{"true"},
+		Refresh:    ddsLtsConfigRefreshFunc(createDdsLtsLogClient, instanceID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      1 * time.Second,
+		MinTimeout: 2 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for associating DDS (%s) with LTS log to become ready: %s ", instanceID, err)
+	}
+
+	return resourceDdsLtsLogRead(ctx, d, meta)
+}
+
+func resourceDdsLtsLogRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+	var (
+		getDdsLtsLogHttpUrl = "v3/{project_id}/instances/logs/lts-configs"
+		getDdsLtsLogProduct = "dds"
+	)
+
+	getDdsLtsLogClient, err := cfg.NewServiceClient(getDdsLtsLogProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	getDdsLtsLogPath := getDdsLtsLogClient.Endpoint + getDdsLtsLogHttpUrl
+	getDdsLtsLogPath = strings.ReplaceAll(getDdsLtsLogPath, "{project_id}", getDdsLtsLogClient.ProjectID)
+
+	getDdsLtsLogResp, err := pagination.ListAllItems(
+		getDdsLtsLogClient,
+		"offset",
+		getDdsLtsLogPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		return diag.Errorf("error retrieving DDS LTS configs: %s", err)
+	}
+
+	getDdsLtsLogRespJson, err := json.Marshal(getDdsLtsLogResp)
+	if err != nil {
+		return diag.Errorf("error marshaling DDS LTS configs: %s", err)
+	}
+
+	var getDdsLtsLogRespBody interface{}
+	err = json.Unmarshal(getDdsLtsLogRespJson, &getDdsLtsLogRespBody)
+	if err != nil {
+		return diag.Errorf("error unmarshaling DDS LTS configs: %s", err)
+	}
+
+	instanceLtsConfig := FilterInstanceLtsConfig(d.Id(), getDdsLtsLogRespBody)
+	instance := utils.PathSearch("instance", instanceLtsConfig, nil)
+	ltsConfigs := utils.PathSearch("lts_configs", instanceLtsConfig, make([]interface{}, 0)).([]interface{})
```

Comment:
suggest to use the following search expression to get LTS config object
```suggestion
	expression := fmt.Sprintf("lts_configs[?id=='%s'] | [0]", d.Id())
	ltsConfigs := utils.PathSearch(expression, getDdsLtsLogRespBody, nil)
```

Reply:
done